packages/zoe/src/contracts/helpers/bondingCurves.js

```diff
@@ -1,102 +1,82 @@
-import Nat from '@agoric/nat';
 import harden from '@agoric/harden';
-import { assert, details } from '@agoric/assert';
 
 import { natSafeMath } from './safeMath';
 
 const { add, subtract, multiply, floorDivide } = natSafeMath;
 
-export const makeConstProductBC = (zoe, issuers) => {
-  const amountMathArray = zoe.getAmountMathForIssuers(issuers);
-  const brands = zoe.getBrandsForIssuers(issuers);
+export const makeConstProductBC = zoe => {
   return harden({
     /**
      * Contains the logic for calculating how much should be given
      * back to the user in exchange for what they sent in. It also
      * calculates the new amount of the assets in the pool. Reused in
-     * several different places, including to check whether an offer is
-     * valid, getting the current price for an asset on user request, and
-     * to do the actual reallocation after an offer has been made.
-     * @param  {amount[]} poolAmountsArray - an array of the current amount in the
-     * liquidity pool
-     * @param  {amount} amountIn - the amount sent in by a user
-     * @param  {number} feeInTenthOfPercent=3 - the fee taken in tenths of
-     * a percent. The default is 0.3%. The fee is taken from amountIn
+     * several different places, including to check whether an offer
+     * is valid, getting the current price for an asset on user
+     * request, and to do the actual reallocation after an offer has
+     * been made.
+     * @param  {extent} inputExtent - the extent of the assets sent in
+     * to be swapped
+     * @param  {extent} inputReserve - the extent in the liquidity
+     * pool of the kind of asset sent in
+     * @param  {extent} outputReserve - the extent in the liquidity
+     * pool of the kind of asset to be sent out
+     * @param  {number} feeBasisPoints=30 - the fee taken in
+     * basis points. The default is 0.3% or 30 basis points. The fee is taken from
+     * inputExtent
      */
-    getPrice: (poolAmountsArray, amountIn, feeInTenthOfPercent = 3) => {
-      Nat(feeInTenthOfPercent);
-      assert(
-        feeInTenthOfPercent < 1000,
-        details`fee ${feeInTenthOfPercent} is not less than 1000`,
-      );
-      const oneMinusFeeInThousandths = subtract(1000, feeInTenthOfPercent);
+    getPrice: ({
+      inputExtent,
+      inputReserve,
+      outputReserve,
+      feeBasisPoints = 30,
+    }) => {
+      const feeTenthOfPercent = floorDivide(feeBasisPoints, 10);
```

Comment:
Why are you normalizing from basis points to tenths of a percent? Please just stick with basis points unless there's a reason.

Reply:
I was thinking we'd be less likely to hit the MAX_SAFE_INTEGER if we were only multiplying by 1000 rather than 10000, but maybe we don't need to worry about that?

Reply:
Interesting point. However, we should transition Nat fully to BigInt before it becomes an issue.